docs/reference/modules/error.md

```diff
@@ -0,0 +1,53 @@
+# `@apostrophecms/error`
+
+This module establishes an error method used throughout Apostrophe code to help format code errors consistently.
+
+::: note
+`apos.error` is the error construction method, not an alias to this module.
+:::
+
+## Related documentation
+
+- [HTTP module error codes](/reference/modules/http.md#adderrors)
+
+## Featured methods
+
+### `error(name, message, data)`
+
+Returns an Error object suitable to throw. The `name` argument will
+be applied to the Error object. Certain values of `name` match to certain HTTP status codes. [See the `@apostrophecms/http` module.](/reference/modules/http.md#adderrors)
+
+`message` may be skipped completely, or included as a string for a longer description. `data` is optional and may contain data about the error, safe to share with an untrusted client.
+
+If the error is caught by Apostrophe's `apiRoute` or `restApiRoute` mechanism, and `name` matches to a status code, an appropriate HTTP error is sent, and `data` is sent as a JSON object, with `message` as an additional property if present. If the `name` argument does not match a code, a general 500 error is sent to avoid disclosing inappropriate information and the error is only logged by Apostrophe server-side.
```

Comment:
The module initialization function is called `apiRoutes`, not `apiRoute`, so best not to use it in the singular especially in code font. I suggest linking to the documentation of those sections.

Throwing errors like this also works in `renderRoutes`.